pkg/distributor/distributorerror/errors.go

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributorerror
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+const (
+	// 529 is non-standard status code used by some services to signal that "The service is overloaded".
+	StatusServiceOverloaded = 529
+)
+
+// ReplicasNotMatch is an error stating that replicas do not match.
+type ReplicasNotMatch struct {
+	replica, elected, format string
+}
+
+// ReplicasNotMatchErrorf formats and returns a ReplicasNotMatch error.
+// The error message is formatted according to the given format specifier, replica and elected.
+// In order to print the limit and the burst, the format specifier must contain two %s verbs.
+func ReplicasNotMatchErrorf(format, replica, elected string) ReplicasNotMatch {
+	return ReplicasNotMatch{
+		replica: replica,
+		elected: elected,
+		format:  format,
+	}
+}
```

Comment:
I don't like the fact that this has to take a format that is compatible with the other params provided. Can we just take the `msg string` and assume that the entity that defined the `format` would format it?

Reply:
Same applies to the rest of the errors in this package.